graphene_django/converter.py

```diff
@@ -313,14 +361,87 @@ def dynamic_type():
 @convert_django_field.register(models.OneToOneField)
 @convert_django_field.register(models.ForeignKey)
 def convert_field_to_djangomodel(field, registry=None):
+    from graphene.utils.str_converters import to_snake_case
+    from .types import DjangoObjectType
+
     model = field.related_model
 
     def dynamic_type():
         _type = registry.get_type_for_model(model)
         if not _type:
             return
 
-        return Field(
+        class CustomField(Field):
+            def wrap_resolve(self, parent_resolver):
+                """
+                Implements a custom resolver which go through the `get_node` method to ensure that
+                it goes through the `get_queryset` method of the DjangoObjectType.
+                """
+                resolver = super().wrap_resolve(parent_resolver)
+
+                # If `get_queryset` was not overridden in the DjangoObjectType,
+                # we can just return the default resolver.
+                if (
+                    _type.get_queryset.__func__
+                    is DjangoObjectType.get_queryset.__func__
+                ):
+                    return resolver
+
+                def custom_resolver(root, info, **args):
+                    # Note: this function is used to resolve FK or 1:1 fields
```

Comment:
Just out of curiosity, if this is used to resolve 1:1 fields, why does `convert_onetoone_field_to_djangomodel` also need this similar `custom_resolver` overriding logic separately?

Reply:
Good question; for each relation kind, we need to deal with both directions: forward (`OneToOneField`, `ForeignKey`, `ManyToManyField`) and backward (`ManyToManyRel`, `ManyToOneRel`, `OneToOneRel`). The resolving logic depends on which side you're working on, hence the separation.